### PR TITLE
feat(api-client):  handling multipart/form-data Content-Type

### PR DESCRIPTION
**Pull request by mkucmus**
### Description

closes: #863 #1319 

the PR also provides a `customized-products` example as a nuxt project (migration was done).

### Type of change

New feature (non-breaking change which adds functionality)

### ToDo's

- [x] Changeset file provided [read more](https://github.com/shopware/frontends/blob/main/CONTRIBUTION.md#changelog-preparation) -->

### CODE REVIEW

File: packages/api-client/src/createAPIClient.ts [hunk 1]
```diff
@@ -154,13 +154,23 @@ export function createAPIClient<
       ...(currentParams.fetchOptions || {}),
     };
 
+    let mergedHeaders = defu(currentParams.headers, defaultHeaders);
```
Comment (mkucmus) [inline packages/api-client/src/createAPIClient.ts:157]: the interesting line here: `defu` accepts the defaults as a **last** parameter. lesson learnt 😂

Comment (BrocksiNet) [reply packages/api-client/src/createAPIClient.ts:157]: Yeah, there is no consequent style that works for all languages and tools.
I always look this up ... so much wasted time :joy:

File: packages/api-client/src/createApiClient.test.ts [hunk 2]
```diff
@@ -340,7 +374,7 @@ describe("createAPIClient", () => {
     controller.abort();
 
     expect(request).rejects.toThrowErrorMatchingInlineSnapshot(
-      `[FetchError: [GET] "${baseURL}context": <no response> This operation was aborted]`,
+      `[FetchError: [GET] "${baseURL}context": <no response> The operation was aborted.]`,
```
Comment (mkucmus) [inline packages/api-client/src/createApiClient.test.ts:377]: TODO: #1324

File: packages/api-client/src/createApiClient.test.ts [hunk 3]
```diff
@@ -256,6 +256,40 @@ describe("createAPIClient", () => {
     );
   });
 
+  it("should remove multipart/form-data headers in case of browser", async () => {
+    // @vitest-environment jsdom
```
Comment (mkucmus) [inline packages/api-client/src/createApiClient.test.ts:260]: @TODO: #1324

**Review by BrocksiNet — APPROVED** [review]

**Pull request merged**